### PR TITLE
Fix resize() so that it respects max_size_max

### PR DESCRIPTION
**Pull request by skeetsaz**
https://github.com/OpenCyphal/CETL/issues/80

### CODE REVIEW

File: include/cetl/variable_length_array.hpp [hunk 1]
```diff
@@ -1424,18 +1425,10 @@ class VariableLengthArray : protected VariableLengthArrayBase<T, Allocator>
...
             return;
         }
 
-        if (nullptr == push_back_impl(std::move(value)))
-        {
-#if defined(__cpp_exceptions)
-            throw std::length_error("size is at capacity. Use reserve to grow the capacity.");
-#endif
-        }
+        push_back_impl(std::move(value));
```
Comment (thirtytwobits) [inline include/cetl/variable_length_array.hpp:1431]: `std::forward`

File: cetlvast/suites/unittest/test_variable_length_array_bool.cpp [hunk 2]
```diff
@@ -335,6 +335,38 @@ TYPED_TEST(VLABoolTests, TestBoolResizeOneBit)
     ASSERT_EQ(0, array[4]);
 }
 
+TYPED_TEST(VLABoolTestsVLAOnly, TestBoolExceedingMaxSizeMax)
```
Comment (thirtytwobits) [inline cetlvast/suites/unittest/test_variable_length_array_bool.cpp:338]: Can this be part of `VLABoolTests`? If not then what's the difference between vector and VLA here?

Comment (skeetsaz) [reply cetlvast/suites/unittest/test_variable_length_array_bool.cpp:338]: No.  This test makes use of max_size_max which std::vector does not have.

File: include/cetl/variable_length_array.hpp (hunk 1 above)
Comment (skeetsaz) [reply include/cetl/variable_length_array.hpp:1431]: Fixed

**Review by thirtytwobits — APPROVED** [review]

**Pull request merged**